robot/impl/resource_manager.go

```diff
@@ -586,11 +586,11 @@ func (manager *resourceManager) completeConfig(
 					gNode.SetLastError(errors.Wrap(err, "resource build error"))
 					return
 				}
-				// if the ctxWithTimeout has an error then that means we've timed out. This means
-				// that resource generation is running async, and we don't currently have good
-				// validation around how this might affect the resource graph. So, we avoid updating
-				// the graph to be safe.
-				if ctxWithTimeout.Err() != nil {
+				// if the ctxWithTimeout fails with DeadlineExceeded, then that means that
+				// resource generation is running async, and we don't currently have good
+				// validation around how this might affect the resource graph. So, we avoid
+				// updating the graph to be safe.
+				if ctxWithTimeout.Err() == context.DeadlineExceeded {
```

Comment:
We want to continue updating the graph because we need resources to be in the graph so we can stop them. Without updating, I ran into a case where, when I sigtermed before the first Reconfigure was completed, a data manager wasn't put in the graph and so wasn't stopped in `robot.Close()` and so was leaking its `uploadData` goroutine. By still updating the graph, the resource manager becomes aware of the data manager (or any other resource with a goroutine) and can stop it appropriately.

Reply:
Ah so sigterming a robot will no longer stop building goroutines?

Reply:
I'm not aware that we've fundamentally changed how goroutines building works here, I think it's possible I ran into a race where the ctx was canceled in one thread and in another one we started up a goroutine for the data manager, but then because the ctx was canceled in the first thread the data manager was never added to the resource graph? The sigterm will definitely kill the data manager goroutine provided the resource manager knows about it, but that requires updating the graph.